### PR TITLE
HIVE-23824

### PR DESCRIPTION
**Pull request by szlta**
## NOTICE

Please create an issue in ASF JIRA before opening a pull request,
and you need to set the title of the pull request which starts with
the corresponding JIRA issue number. (e.g. HIVE-XXXXX: Fix a typo in YYY)
For more details, please see https://cwiki.apache.org/confluence/display/Hive/HowToContribute

### CODE REVIEW

File: llap-client/src/java/org/apache/hadoop/hive/llap/io/api/LlapIo.java [hunk 1]
```diff
@@ -35,6 +41,15 @@
    */
   long purge();
 
+  /**
+   * Returns a deserialized OrcTail instance associated with the ORC file on the given path.
+   *  Raw content is either obtained from cache, or from disk if there is a cache miss.
+   * @param path Orc file path
+   * @param conf jobConf
```
Comment (pvary) [inline llap-client/src/java/org/apache/hadoop/hive/llap/io/api/LlapIo.java:48]: Missing javadoc for cachetag

File: llap-server/src/java/org/apache/hadoop/hive/llap/io/encoded/OrcEncodedDataReader.java [hunk 2]
```diff
@@ -575,31 +626,8 @@ private OrcFileMetadata getFileFooterFromCacheOrDisk() throws IOException {
...
-              totalSize += buf.getByteBufferRaw().remaining();
-            }
-            bb = ByteBuffer.allocate(totalSize);
-            for (MemoryBuffer buf : tailBufferArray) {
-              bb.put(buf.getByteBufferDup());
-            }
-            bb.flip();
-          }
-          OrcTail orcTail = ReaderImpl.extractFileTail(bb);
+          OrcTail orcTail = getOrcTailFromLlapBuffers(tailBuffers);
```
Comment (pvary) [inline llap-server/src/java/org/apache/hadoop/hive/llap/io/encoded/OrcEncodedDataReader.java:629]: What happens if the tail is evicted from the cache in the meantime (after the check, but before this line). Or it is locked during this code?

Comment (pvary) [reply llap-server/src/java/org/apache/hadoop/hive/llap/io/encoded/OrcEncodedDataReader.java:629]: Oh, I see now... it is decRefBuffer...

Comment (szlta) [reply llap-server/src/java/org/apache/hadoop/hive/llap/io/encoded/OrcEncodedDataReader.java:629]: Yes, it won't be evicted as it is locked. The getFileMetadata invocation will incRef it (if it is found of course)

**Pull request merged**